util/src/main/scala/geotrellis/util/Haversine.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2017 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.util
+
+object Haversine {
+  val EARTH_RADIUS = 6378137d // Use what gdal2tiles uses.
+
+  // (x: Double, y: Double) points
+  def apply(start: (Double, Double), end: (Double, Double), R: Double = EARTH_RADIUS): Double = {
```

Comment:
For future GeoTrellisites, I'd suggest a doc comment indicating what the components of `start` and `end` are. e.g. (Latitude, Longitude), or (Longitude, Latitude) or (radius, angle), etc.